app/uk/gov/hmrc/trackingconsentfrontend/assets/src/interfaces/optimizelyCommunicatorFactory.ts

```diff
@@ -5,13 +5,37 @@ const optimizelyCommunicatorFactory = (window: Window): Communicator => ({
   sendPreferences: (userPreferences: UserPreferences) => {
 
     const setOptimizelyOptOut = (isOptOut: Boolean) => {
-      window.optimizely.push({ 'type': 'optOut', isOptOut });
+      window.optimizely.push({ 'type': 'optOut', isOptOut })
     }
 
-    window.optimizely = window.optimizely || [];
+    const setupOptimizelyIfConfigured = () => {
+      const regex = /^(\/|(https:)?\/\/www\.tax\.service\.gov\.uk\/|(https:)?\/\/www\.(qa|dev|staging)\.tax\.service\.gov\.uk\/|(http:)?\/\/localhost:12345\/)tracking-consent/
+      const isTruthy = x => !!x
+      const scriptTags = Array.from(document.getElementsByTagName('script'))
+      const trackingConsentScripts = scriptTags
+        .filter(scriptTag => (scriptTag.getAttribute('src') || '')
+          .match(regex))
+      const optimizelyIDs = trackingConsentScripts
```

Comment:
If we have already determined the current script, we don't need to generate an array of ids, we can just do a getAttribute on the current script. I'm not sure the isTruthy function is required and can be inlined.